ros2/cc_defs.bzl

```diff
@@ -64,7 +64,39 @@ def ros2_c_binary(name, ros2_package_name = None, **kwargs):
     """
     _ros2_cc_target(cc_binary, "c", name, ros2_package_name, **kwargs)
 
-def ros2_cpp_binary(name, ros2_package_name = None, **kwargs):
+def _ros2_cpp_exec(target, name, ros2_package_name = None, set_up_ament = False, **kwargs):
+    if set_up_ament == False:
+        _ros2_cc_target(target, "cpp", name, ros2_package_name, **kwargs)
+        return
+
+    target_impl = name + "_impl"
+    _ros2_cc_target(cc_binary, "cpp", target_impl, ros2_package_name, tags = ["manual"], **kwargs)
+
+    is_test = target == cc_test
+
+    launcher = "{}_launch".format(name)
+    sh_launcher(
+        launcher,
+        deps = [target_impl],
+        template = "@com_github_mvukov_rules_ros2//ros2:launch.sh.tpl",
+        substitutions = {
+            "{entry_point}": "$(rootpath {})".format(target_impl),
+        },
+        tags = ["manual"],
+        data = [target_impl],
+        testonly = is_test,
+    )
+
+    sh_target = native.sh_test if is_test else native.sh_binary
+    sh_target(
+        name = name,
+        srcs = [launcher],
+        data = [target_impl],
+        tags = kwargs.pop("tags", []),
```

Comment:
AFAIK, if you would have tags in kwargs and also have manual tag for _ros2_cc_target, that would fail. The intention was to make sure that _ros2_cc_target only has manual tag. IMO, extraction of `tags` from kwargs should happen before the call to _ros2_cc_target.

Reply:
Ah, yes, you're right! I reverted that change now.